src/components/20-molecules/datepicker/index.js

```diff
@@ -266,13 +266,28 @@ class AXADatepicker extends NoShadowDOM {
 
     this.setMonthAndYearItems(month, year);
 
-    const { width = 'auto', height = '40px', error, invalid } = this;
+    const { width = '100%', height = '40', error, invalid, style } = this;
 
     const formattedStyle = parameter =>
       `${parameter}${/^\d+$/.test(parameter) ? 'px' : ''}`;
 
-    const formattedWidth = `width:${formattedStyle(width)}`;
-    const formattedHeight = `height:${formattedStyle(height)}`;
+    let formattedWidth = `width:${formattedStyle(width)}`;
+    let formattedHeight = `height:${formattedStyle(height)}`;
+
+    // % on html width or height attribute not allowed, so set it as css attribute
+    if (width.indexOf('%') > -1) {
+      style.width = width;
+      formattedWidth = ''; // do not set width to childs in this case
+    } else {
+      style.width = null;
+    }
+
+    if (height.indexOf('%') > -1) {
+      style.height = height;
+      formattedHeight = ''; // do not set height to childs in this case
+    } else {
+      style.height = null;
+    }
```

Comment:
lots of DRY violations. I suggest to rewrite as (tested):
` const formattedStyle = (parameter, dimension) => {
      const isPercentage = parameter.indexOf('%') > -1;
      if (isPercentage) {
        style[dimension] = parameter; // side-effect: set style of host
      }
      // pixel parameters: if 'px' not already given (digits-only value), append it
      return isPercentage
        ? ''
        : `${dimension}:${parameter}${/^\d+$/.test(parameter) ? 'px' : ''}`;
    };

    const formattedWidth = formattedStyle(width, 'width');
    const formattedHeight = formattedStyle(height, 'height');`

Reply:
Thats nice, thanks. But line 289 `style.width = null;` is important to remove the css attribute, if width changes from % to px.

But i will refactor this part of my code.